jsapp/js/components/activity/formActivity.tsx

```diff
@@ -1,7 +1,116 @@
+import type {ReactNode} from 'react';
+import {useState} from 'react';
+import '../../../scss/components/_kobo.form-view.scss';
+import type {KoboSelectOption} from '../common/koboSelect';
+import KoboSelect from '../common/koboSelect';
+import './formActivity.scss';
+import type {UniversalTableColumn} from 'jsapp/js/universalTable/universalTable.component';
+import Button from '../common/button';
+import PaginatedQueryUniversalTable from 'jsapp/js/universalTable/paginatedQueryUniversalTable.component';
+import type {ActivityLogItem} from 'jsapp/js/query/queries/activityLog.query';
+import {useActivityLogQuery} from 'jsapp/js/query/queries/activityLog.query';
+import moment from 'moment';
+
+interface TableDataItem {
+  description: ReactNode;
+  date: ReactNode;
+}
+
+const mockOptions: KoboSelectOption[] = [
+  {value: '1', label: 'Option 1'},
+  {value: '2', label: 'Option 2'},
+  {value: '3', label: 'Option 3'},
+];
+
+const columns: UniversalTableColumn[] = [
+  {
+    key: 'description',
+    label: t('Event description'),
+  },
+  {
+    key: 'date',
+    label: t('Date'),
+    size: 100,
```

Comment:
🤔 After taking a better look at it, I don't think the `cellFormatter` will gonna be enough. The idea of the `rowRenderer` is that it takes the whole object and generate each cell because, in this case, I the cell will need to aggregate multiple values from the object to generate the complex rendering, even adding the Avatar. We could extract the avatar to another cell, but we'd still need to join multiple values.

For the universal table it's easier to implement something like this, but since the paginated takes care of fetching data we need a way to process it, that's why I added types to it, so we can indicate which is the data type and which is the cell type, because I don't think the types will match frequently.

WDYT?

Reply:
Just to close this up: The new version of PaginatedQueryUniversalTable has a different cellFormatter, which takes the whole DataItem as a parameter. I rebased this branch, my addition was removed and I'm now using the cellFormatter.
Thank you.